core/templates/dev/head/dashboard/Dashboard.js

```diff
@@ -34,10 +35,10 @@ oppia.controller('Dashboard', [
     };
 
     $rootScope.loadingMessage = 'Loading';
-    $http.get('/dashboardhandler/data').then(function(response) {
-      var data = response.data;
-      $scope.explorationsList = data.explorations_list;
-      $scope.collectionsList = data.collections_list;
+    DashboardBackendApiService.fetchDashboardData().then(function(response) {
```

Comment:
Just chiming in here: I think a more general solution is out of scope for this PR, but perhaps you should file an issue so this can be brought up at the TL meeting. 

I think handling the promise rejection with a warning in alertsService is probably best here.